### PR TITLE
Onboard vmwarecloudsimple for autogeneration

### PR DESCRIPTION
**Pull request by anthony-c-martin**
```

> azure-schema-generator@1.0.0 generate-single C:\github\azure-resource-manager-schemas\generator
> ts-node cmd/generatesingle "vmwarecloudsimple/resource-manager"

[C:\Users\antmarti\AppData\Local\Temp\schm_azspc] executing: git fsck
[C:\Users\antmarti\AppData\Local\Temp\schm_azspc] executing: git reset -q .
[C:\Users\antmarti\AppData\Local\Temp\schm_azspc] executing: git checkout -q -- .
[C:\Users\antmarti\AppData\Local\Temp\schm_azspc] executing: git clean -q -fd
[C:\Users\antmarti\AppData\Local\Temp\schm_azspc] executing: git gc
[C:\Users\antmarti\AppData\Local\Temp\schm_azspc] executing: git fetch
[C:\Users\antmarti\AppData\Local\Temp\schm_azspc] executing: git checkout -q origin/master
[C:\github\azure-resource-manager-schemas\generator] executing: autorest-beta.cmd --azureresourceschema --azureresourceschema.multi-scope=true --output-folder=C:\Users\antmarti\AppData\Local\Temp\n016j4p0mm --tag=all-api-versions --api-version=2019-04-01 --title=none C:\Users\antmarti\AppData\Local\Temp\schm_azspc\specification\vmwarecloudsimple\resource-manager\readme.md
AutoRest code generation utility [version: 3.0.6116; node: v10.15.3, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      'C:\github\azure-resource-manager-schemas\generator\node_modules\@autorest\core\dist' (3.0.6121)
   Loading local AutoRest extension '@autorest/azureresourceschema' (C:\github\azure-resource-manager-schemas\generator\node_modules\@autorest\azureresourceschema)
[8.62 s] Generation Complete
================================================================================================================================
Filename: C:\Users\antmarti\AppData\Local\Temp\n016j4p0mm\2019-04-01\Microsoft.VMwareCloudSimple.json
Provider Namespace: Microsoft.VMwareCloudSimple
API Version: 2019-04-01
Resource Types (Resource Group Scope):
- dedicatedCloudNodes
- dedicatedCloudServices
- virtualMachines
================================================================================================================================
```

### CODE REVIEW

**Comment by ifeoluwaokunoren** [comment]
Hi @tenmozes,
I see you created this [PR](https://github.com/Azure/azure-resource-manager-schemas/pull/725)
To simply the schema generation process, ARM is working on an auto-generation pipeline your team can onboard to. 
This means once a swagger PR from the [Azure REST-API spec repo](https://github.com/Azure/azure-rest-api-specs/tree/master/specification) is merged either for a new api-version or an update to an existing api-version – the auto generation pipeline would generate the required schemas for that api-version daily – so changes in swagger would be automatically picked up.
The required action is  for you to kindly review this PR to ensure that all expected resources have been detected and generated as expected.

File: schemas/2019-04-01/Microsoft.VMwareCloudSimple.json [hunk 1]
```diff
@@ -475,17 +530,18 @@
...
-              "$ref": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+              "$ref": "https://schema.management.azure.com/schemas/common/definitions.json#/definitions/expression"
             }
           ],
           "description": "The list of Virtual VSphere Networks"
         }
       },
       "required": [
         "amountOfRam",
+        "guestOS",
```
Comment (tenmozes) [inline schemas/2019-04-01/Microsoft.VMwareCloudSimple.json]: guestOS and guestOSType are not mandatory anymore,
templateId is mandatory for VM creation

**Comment by tenmozes** [comment]
Hi guys, thank you for adding VMWareCloudsimple to autogeneration,
i have one question
  we have an entity Virtual Machine and there are two ways how to create it from Azure Portal and its should appear in VCenter, from VCenter and it should appear on Portal (via hydration mechanism). From Azure side as for now, we can create a virtual machine only from the template (that's why templated is mandatory for creation). From VCenter there are many ways to create machine (clone from another one, use content library, sync from another VCenter). For an update we don' t require templateId. 
How can we reflect this validation in our specification and don't add manually templateId to required fields and use autogeneration for deployment templates?

**Comment by tenmozes** [comment]
thank you so much guys, very appreciate it, autogeneration will save my time :)

File: schemas/2019-04-01/Microsoft.VMwareCloudSimple.json (hunk 1 above)
Comment (anthony-c-martin) [reply schemas/2019-04-01/Microsoft.VMwareCloudSimple.json]: I've re-run this to pick up the latest swagger changes - commit https://github.com/Azure/azure-resource-manager-schemas/pull/726/commits/76385836159b8c4e9fdb8f38bdd48abf626ae5f3 shows the diff.

Comment (tenmozes) [reply schemas/2019-04-01/Microsoft.VMwareCloudSimple.json]: thank you, looks perfect

**Pull request merged**